templates/ecs.yaml

```diff
@@ -0,0 +1,458 @@
+    Type: String
+    Description: The environment that the Flask application is running in. Valid values are `production`, `staging`, `development`, or `test`.
+    Default: 'production'
+    AllowedValues:
+      - test
+      - development
+      - staging
+      - production
+  DbHost:
+    Type: String
+    Description: The reader endpoint for the DB cluster. For example - mystack-mydbcluster-ro-123456789012.us-east-2.rds.amazonaws.com
+    NoEcho: true
+  DbPort:
+    Type: Number
+    Description: The port of the DB cluster.
+    Default: 5432
+    NoEcho: true
+  DbAccessSecurityGroup:
+    Type: String
+    Description: The DB Security Group.
+  DbCredsArn:
+    Description: The ARN of the Database secret in the Secret Manager
+    NoEcho: 'true'
+    Type: String
+  KmsKeyArn:
+    Description: 'Encryption key ARN to use for storage'
+    Type: String
+Resources:
+  ########################
+  ### Managed Policies ###
+  ########################
+  # This is necessary so it can read the Secrets Manager entry that contains the external registry credentials.
+  AccessSecretPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Action:
+              - kms:Decrypt
+              - secretsmanager:GetSecretValue
+              - ssm:GetParameters
+            Resource:
+              - !Ref 'KmsKeyArn'
+              - !Ref 'DbCredsArn'
+              - !Sub 'arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:iatlas_gitlab_registry_creds-??????'
+              - !Sub 'arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/*'
+          - Effect: Allow
+            Action:
+              - logs:CreateLogStream
+            Resource: '*'
+  EcsServicePolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Action:
+              - 'elasticloadbalancing:DeregisterInstancesFromLoadBalancer'
+              - 'elasticloadbalancing:DeregisterTargets'
+              - 'elasticloadbalancing:Describe*'
+              - 'elasticloadbalancing:RegisterInstancesWithLoadBalancer'
+              - 'elasticloadbalancing:RegisterTargets'
+              - 'ec2:Describe*'
+              - 'ec2:AuthorizeSecurityGroupIngress'
+            Resource: '*'
+  Ec2Policy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Action:
+              - 'ecs:CreateCluster'
+              - 'ecs:DeregisterContainerInstance'
+              - 'ecs:DiscoverPollEndpoint'
+              - 'ecs:Poll'
+              - 'ecs:RegisterContainerInstance'
+              - 'ecs:StartTelemetrySession'
+              - 'ecs:Submit*'
+              - 'logs:CreateLogStream'
+              - 'logs:PutLogEvents'
+            Resource: '*'
```

Comment:
shouldn't this be scoped to the `EcsService`?

Reply:
@zaro0508 - When I scoped this to the `EcsService`, it fails. The Task don't run in the Cluster.